packages/wonder-blocks-icon/components/icon.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+import React from "react";
+import {StyleSheet} from "aphrodite";
+import Color from "@khanacademy/wonder-blocks-color";
+
+import addStyle from "../../wonder-blocks-core/util/add-style.js";
+import iconPaths from "../util/icon-paths.js";
+
+type IconName = $Keys<typeof iconPaths>;
+type IconSize = "small" | "medium" | "large" | "xlarge";
+
+type Props = {
+    /**
+     * Passed transparently to the SVG. If not provided, we assume the SVG is
+     * purely decorative and it is invisible to screenreaders.
+     */
+    "aria-label"?: string,
+    /**
+     * The color of the icon. Will default to off-black.
+     */
+    color?: string,
+    /**
+     * One of our named icons from icon-paths.js
+     */
+    name: IconName,
+    /**
+     * One of `small` (16px), `medium` (24px), `large` (48px),
+     * or `xlarge` (96px).
+     */
+    size: IconSize,
+    /**
+     * Styles that can be processed by `addStyle` — bare style objects,
+     * Aphrodite style objects, or arrays thereof.
+     */
+    style?: any,
+};
+
+const StyledSVG = addStyle("svg");
+
+const viewportPixelsForSize: {[IconSize]: number} = {
+    small: 16,
+    medium: 24,
+    large: 48,
+    xlarge: 96,
+};
+
+const getPathForIcon = (
+    name: IconName,
+    size: IconSize,
+): {
+    assetSize: IconSize,
+    path: string,
+} => {
+    if (!iconPaths[name]) {
+        throw new Error(`There is no icon named ${name}.`);
+    }
+    const paths = iconPaths[name];
+    if (paths && paths[size]) {
+        // Great, we have the IconSize we actually requested
+        return {assetSize: size, path: paths[size]};
+    } else {
+        // Oh, no, we don't have the right IconSize! Let's find the next best
+        // one...we prefer
+        const desiredPixelSize = viewportPixelsForSize[size];
+        const availableSizes = Object.keys(paths);
+        const assetSizes = availableSizes.sort((availableSize) => {
+            const availablePixelSize = viewportPixelsForSize[availableSize];
+            const sizeDifference = availablePixelSize - desiredPixelSize;
+            return sizeDifference < 0 ? sizeDifference : 1;
+        });
+        const bestAssetSize = assetSizes[0];
+        if (bestAssetSize && paths[bestAssetSize]) {
+            return {assetSize: bestAssetSize, path: paths[bestAssetSize]};
+        } else {
+            throw new Error(
+                `No path available for icon ${name} at size ${size}`,
+            );
+        }
+    }
+};
+
+/**
+ * An Icon displays a small informational or decorative image as an SVG. Wonder
+ * Blocks has a fixed set of icons and icon sizes.
+ *
+ * ```js
+ * <Icon
+ *     name={correct}
+ *     color={Colors.green}
```

Comment:
🛑 We should change the color to be something that we want to support (like Color.white).